synapse/app/event_creator.py

```diff
@@ -192,6 +190,8 @@ def start(config_options):
         database_engine=database_engine,
     )
 
+    setup_logging(ss, config, use_worker_options=True)
```

Comment:
What happens if we log before this? I assume that for some reason we can't set up structured logging before this? I'm mainly worried that we'll end up losing any logging that happens while we create the server and set up the database.

Reply:
if you are logging to the twisted logging system, they go into the `globalLogBeginner`, which stores logs between init/import of twisted and then outputs them when the logging system starts. if you are logging to the stdlib logger before this, they go into the void

(so we should use the twisted logger, so they go somewhere reasonable)

Reply:
Oh, interesting. So if we use the stdlib config we don't get those log lines, but if we use the new structured config we don't?

Reply:
if we use the structured config we will, yes